general-concepts/dependencies/text.xml

```diff
@@ -850,5 +850,309 @@ USE flags. This would then break building your ebuild.
+</p>
+
+<codesample lang="ebuild">
+# Bad:
+RDEPEND="
+	|| (
+		dev-libs/A:=
+		dev-libs/B:=
+	)
+"
+
+# Better:
+RDEPEND="
+	|| (
+		dev-libs/A
+		dev-libs/B
+	)
+	dev-libs/A:=
+"
+</codesample>
+
+<note>
+The above is only an illustration of structure. Choose the package for the
+slot-operator binding that truly determines ABI compatibility for your
+package, and constrain its version range appropriately.
+</note>
+
+</body>
+</subsection>
+
+<subsection>
+<title>Any-of and <c>:*</c> across classes</title>
+<body>
+
+<p>
+Any-of groups (<c>|| ( ... )</c>) and the <c>:*</c> slot operator are valid in
+all dependency classes. However, there is no binding between occurrences in
+different classes. An any-of in <c>DEPEND</c> guarantees only that at least one
+alternative is installed before the build; an any-of in <c>RDEPEND</c> or
+<c>PDEPEND</c> guarantees only that at least one is installed for runtime.
+You should not assume the same alternative or slot will be used for both.
+</p>
+
+</body>
+</subsection>
+
+<subsection>
+<title>The <c>:=</c> slot operator across classes</title>
+<body>
+
+<p>
+The <c>:=</c> operator is technically valid in all classes but is useful only
+when the matching package is installed at the time metadata are recorded
+(install from source or binary package creation). Practically, the atom using
+<c>:=</c> should be present in <c>RDEPEND</c> to express the rebuild
+relationship, and <c>DEPEND</c> must guarantee that a matching package is
+installed at the relevant time.
+</p>
+
+<codesample lang="ebuild">
+# Typical pattern — ensure the provider is present at build time and tie
+# rebuilds to its sub-slot changes
+DEPEND="dev-libs/foo:="
+RDEPEND="dev-libs/foo:="
+</codesample>
+
+<note>
+The atoms need not be textually identical as long as <c>DEPEND</c> guarantees
+that some package matching the <c>RDEPEND</c> <c>:=</c> atom is installed when
+metadata are recorded.
+</note>
+
+</body>
+</subsection>
+
+<subsection>
+<title>Blockers across dependency classes</title>
+<body>
+
+<p>
+Blockers are valid in all classes but their usefulness differs:
+</p>
+
+<ul>
+  <li>
+    Weak blockers (<c>!</c>) are primarily meaningful in <c>RDEPEND</c>, where
+    the uninstall of the blocked package may be delayed until after the new
+    package starts installing, allowing file collisions to be replaced.
+    Technically they are allowed in <c>DEPEND</c>, but they do not influence
+    the build environment and are not useful there on their own.
+  </li>
+  <li>
+    Strong blockers (<c>!!</c>) must be resolved before the dependency is
+    considered satisfied. They therefore make sense in both <c>DEPEND</c>
+    (before building) and <c>RDEPEND</c> (before installation).
+  </li>
+  <li>
+    In <c>PDEPEND</c>, weak blockers act effectively like those in
+    <c>RDEPEND</c>, while strong blockers behave more like weak ones because
+    satisfaction may be delayed post-install.
```

Comment:
I don't think that's true, is it?

Reply:
I think we should drop it. I'm not convinced it's guaranteed.